system/lib/wasmfs/syscalls.cpp

```diff
@@ -17,50 +17,48 @@
 #include <stdarg.h>
 #include <stdlib.h>
 #include <sys/stat.h>
+#include <unistd.h>
 #include <utility>
 #include <vector>
 #include <wasi/api.h>
-#include <unistd.h>
 
 extern "C" {
 
 using namespace wasmfs;
 
 // Copy the file specified by the pathname into JS.
 // Return a pointer to the JS buffer in HEAPU8.
+// Caller must free the returned pointer.
 void* emscripten_wasmfs_read_file(char* path) {
   struct stat file;
   int err = 0;
   err = stat(path, &file);
   if (err < 0) {
-    emscripten_console_error(
-      "Fatal error in FS.readFile");
+    emscripten_console_error("Fatal error in FS.readFile");
     abort();
   }
-  
-  size_t size = file.st_size;
+
+  // file.st_size could exceed uint32_t in wasm64.
+  assert(file.st_size < std::numeric_limits<uint32_t>::max());
```

Comment:
Should this be a runtime error instead of an assert? Asserts make sense for internal logic, but if this is invalid input by the user, probably we want to error on it, similar to the abort right before it.